<p>
<br/><br/>
<img src="https://github.com/stjudeDNBBinfCore/Trainings/figures/img/DNB-BINF-Core-logo.png"  width="500" >
<br/><br/>
</p>

# Intro to sc/snRNA-seq and Advanced visualization with R Shiny Workshop

The `Intro to sc/snRNA-seq and Advanced visualization with R Shiny` provides an introduction to QC of raw sc/snRNA-seq data, QC of sc/snRNA-seq alignment data, and cell-leveling filtering of sc/snRNA-seq data. Additionally, this course demonstrates how R Shiny applications can be utilized to explore sc/snRNA-seq data and produce figures for sc/snRNA-seq analysis.
 
 
For the visualization portion of the workshop, we utilize the [Honnell et al., 2022](https://doi.org/10.1038/s41467-021-27924-y) cohort displayed in the [R Shiny app](https://vsx2-deletion.stjude.org/).

 
## Below is the main directory structure listing the analyses and data files used in this repository

```
├── gitignore.txt
├── lectures
|   └── Intro-to-sc-snRNA-seq-2024-09-18.pptx
└── README.md
└── resources
```

## Course Authors

Sharon Freshour, PhD ([@sharonfreshour](https://github.com/sharonfreshour))

## Contact

Contributions, issues, and feature requests are welcome! Please feel free to check [issues](https://github.com/stjudeDNBBinfCore/Trainings/issues).

---

*These materials have been developed by the Bioinformatic core team at the [St. Jude Children's Research Hospital](https://www.stjude.org/). These are open access materials distributed under the terms of the [BSD 2-Clause License](https://opensource.org/license/bsd-2-clause), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.*