<p align="center">
  <img src="figures/img/ScRNASeqSnap_logo.png" alt="snap-logo" width="560px" />
</p>
<p align="center">
  <a href="https://github.com/stjude-dnb-binfcore/trainings/blob/main/LICENSE"><img src="https://img.shields.io/github/license/stjude-dnb-binfcore/kf-template-repo.svg?style=for-the-badge"></a>
</p>



# Single cell RNA Seq Snap workflow (ScRNASeqSnap) Workshop

The `Single cell RNA Seq Snap workflow (ScRNASeqSnap)` is an introduction to the upstream and downstream analyses modules available for analyzing single cell and single nuclei RNA (sc/snRNA) data from 10X sequencing technology. The [Single cell RNA Seq Snap workflow (ScRNASeqSnap)](https://github.com/stjude-dnb-binfcore/sc-rna-seq-snap) is an initiative of the [Bioinformatics Core at the Department of Developmental Neurobiology at the St. Jude Children's Research Hospital](https://www.stjude.org/research/departments/developmental-neurobiology/shared-resources/bioinformatic-core.html). 

In this workshop, we will discuss tools and methods available in the repo and walk you through results, plots, and reports generated during the testing phase of the code review process.
 

| Audience | Computational Skills | Prerequisites | Duration |
:----------|:----------|:----------|:----------|
| Biologists/Bioinformaticians | Beginner/Intermediate | Shell basics | 1.5h|


## Learning Objectives

1. ....
2. 
3. 
4. 
   

## Target Audience
The course is intended for cancer informatics tool developers, particularly those creating tools as a part of the [Department of Neurobiology at the St. Jude Children's Research Hospital](https://www.stjude.org/research/departments/developmental-neurobiology.html), but also for wet lab biologists that are interested in exploring the capabilities of the `ScRNASeqSnap` pipeline.

> These materials were developed for a trainer-led workshop, but are also amenable to self-guided learning.

## Lessons
* [sc-rna-seq-snap-repo-2024-12](https://github.com/stjude-dnb-binfcore/trainings/tree/main/courses/sc-rna-seq-snap-repo/lectures/sc-rna-seq-snap-repo-2024-12.pdf)


## Access to the workshop recording

St. Jude employees can have access to the [recording from the workshop]().




> Pre-requisite: Introduction to Shell and GitHub

This workshop focuses on teaching how to navigate the analysis modules, discuss the methods utilized per module, and the reports/output files expected. We will also disucss best and effective practices for reproducibility and code review in computational biology teams that our team leveraged during the pipeline development process.

The material in this workshop will assume participants have some practical experience using `git` and `GitHub`. However, users with now computational skills or prior experience are welcome.

Materials:
* `./resources`: supplemental materials and learning resources 
* `./lectures`: slides covering the `Learning Objectives`.

 
## Below is the main directory structure listing the analyses and data files used in this repository

```
├── lectures
|   ├── sc-rna-seq-snap-repo-2024-12.pdf
|   └── sc-rna-seq-snap-repo-2024-12.pptx
├── resources
|   ├── ...
|   └── ...
└── README.md
```

## Course Authors

Antonia Chroni, PhD ([@AntoniaChroni](https://github.com/AntoniaChroni))

## Contact

Contributions, issues, and feature requests are welcome! Please feel free to check [issues](https://github.com/stjude-dnb-binfcore/trainings/issues).

---

*These materials have been developed by the Bioinformatic core team at the [St. Jude Children's Research Hospital](https://www.stjude.org/). These are open access materials distributed under the terms of the [BSD 2-Clause License](https://opensource.org/license/bsd-2-clause), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.*