---
title: "Tutorial on how to use the Single cell RNA Seq Snap pipeline"
date: "`r Sys.Date()`"
papersize: a4
fontsize: 11pt
links-as-notes: true
output:
  html_document:
    toc: TRUE
    toc_float: TRUE
    df_print: paged
    code_folding: hide
    toc_depth: 2
    highlight: tango
    number_sections: TRUE
  pdf_document:
    toc: TRUE
    highlight: tango
    number_sections: TRUE
    latex_engine: lualatex
    keep_tex: FALSE
    fig_caption: yes
    fig_crop: no
    fig_height: 2
    fig_width: 3
    toc_depth: 2
always_allow_html: TRUE
urlcolor: blue
linkcolor: black
citecolor: blue
geometry: margin=1in
header-includes: 
  - \usepackage{titling}
  - \usepackage{fancyhdr}
  - \usepackage{graphicx}
  - \usepackage{float}
params:
  root_dir: './'
---

```{r logo-file, echo=FALSE}
attach(params)
# Insert logo on the top of the html report 
logo_file <- file.path(root_dir, "img", "DNB-BINF-Core-logo.png")
htmltools::img(src = knitr::image_uri(logo_file), alt = "logo", style = "position:absolute; top:0; left:0; padding:0px; height:120px;")
detach(params)
```


\addtolength{\headheight}{2.0cm} 
\fancypagestyle{plain}{} 
\thispagestyle{fancy}
\fancyhead[L]{\includegraphics[height=120px]{`r logo_file`}}
\renewcommand{\headrulewidth}{0pt}

<style type="text/css">
:root {--DNB_BINF_Core_color: #00427B;}

h1.title {margin-top: 130px;
          margin-bottom: 25px;
          font-size: 36px;}

.nobullet li {list-style-type: none;}

.reporthead {font-size: 20px;}

body { /* Normal */
  font-size: 16px;
  font-style: Arial, Helvetica, sans-serif;}

h1 {color: var(--DNB_BINF_Core_color);
    font-size: 28px;
    margin-top: 50px;}

h2 {color: var(--DNB_BINF_Core_color);
    font-size: 20px;}

.list-group-item.active, .list-group-item.active:focus, .list-group-item.active:hover {
  background-color: var(--DNB_BINF_Core_color);}
</style>

<a href="https://wiki.stjude.org/display/CAB">

</a>

\pagebreak

# Introduction

## Overview

This notebook offers guidelines and best practices for effectively using the [Single cell RNA Seq Snap pipeline](https://github.com/stjude-dnb-binfcore/sc-rna-seq-snap). Please note that we will continue to maintain and update the pipeline with new analysis modules.

## Prerequisites

We assume users have some practical experience using Shell, `git` and `GitHub`.


# How to access and run the snap pipeline

We recommend that users fork the `sc-rna-seq-snap` repo. Team members should use the [stjude-dnb-binfcore](https://github.com/stjude-dnb-binfcore) account, while others can use their preferred GitHub account. We welcome collaborations, so feel free to reach out if you're interested in being added to the `stjude-dnb-binfcore` account.

## Naming of the repo/project

  - Team members: keep the name of the pipeline and add the name of the project at the end, e.g., `sc-rna-seq-snap-Vsx2_SE`.
  - Others: keep the name of the pipeline as it is, unless you plan to use the repo for multiple projects. Then, you should follow the same approach as recommended for team members.

## Privacy repo/project

Please note that the forked repositories are public and contain the code from the main branch of the `sc-rna-seq-snap` repo (code reviewed and tested). User needs to ensure that the main branch of the forked repo is always up to date with `stjude-dnb-binfcore/sc-rna-seq-snap:main`. 

No results will be published unless the project is under review or has been officially published. 

Furthermore, we adhere to strict guidelines to ensure privacy and protect sensitive data. Sensitive file paths and patient-related data must never be published on GitHub.

## Accessing the code

Next, open the terminal, navigate to your desired directory, and use `git clone` to clone the repository. For more details, see the [README.md](https://github.com/stjude-dnb-binfcore/sc-rna-seq-snap/blob/main/README.md).


# Data

## Set up parameters for the project

Replace the `project_parameters.Config.yaml` with the appropriate file paths and parameters specific to your project.

## Genome references

Our team at the Bioinformatics core at DNB maintains the following genome references: `GRCh38`, `GRCm39`, `GRCh38ANDGRCm39`, and `GRCh38_GFP_tdTomato`, for human, mouse, and dual index genomes, respectively. Please submit an [issue](https://github.com/stjude-dnb-binfcore/sc-rna-seq-snap/issues) to request the path to the reference genome of preference. 


## Cell type gene marker lists 

Our team at the Bioinformatics core at DNB maintains the following cell type gene marker lists for cell type annotation: `mouse brain tissue`, `human retina tissue` and `mouse retina tissue`. Please submit an [issue](https://github.com/stjude-dnb-binfcore/sc-rna-seq-snap/issues) to request the list of gene markers of preference. 


# Analysis

## Analysis module

Please refer to the `<analysis_module>/README.md` files for instructions on how to run the specific analysis module. These files contain the required parameters and necessary files to successfully execute the moduleâ€™s pipeline.

## Container Overview

We have generated a Docker image that contain all tools, packages, and dependencies necessary to run the code and analyses modules. The environment is specifically configured for `Rstudio/R v4.4.0` and `Seurat v4.4.0`. For more details, please refer to the [README.md](https://github.com/stjude-dnb-binfcore/sc-rna-seq-snap/blob/main/run-container/README.md).

## CPU and Memory Resources

While we have provided estimates for the computational resources required, users may need to adjust memory settings based on the size of the cohort and the specific demands of the analysis. For reference, the estimated resources are for 8 samples with approximately 50,000 cells in total. For further details, please consult the [README.md](https://github.com/stjude-dnb-binfcore/sc-rna-seq-snap/blob/main/README.md).


# Contact

Contributions, issues, and feature requests are welcome! Please feel free to check [issues](https://github.com/stjude-dnb-binfcore/trainings/issues).

## Authors

Antonia Chroni, PhD ([@AntoniaChroni](https://github.com/AntoniaChroni))

---

*These materials have been developed by the Bioinformatic core team at the [St. Jude Children's Research Hospital](https://www.stjude.org/). These are open access materials distributed under the terms of the [BSD 2-Clause License](https://opensource.org/license/bsd-2-clause), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.*


```{r include=FALSE}
sessionInfo()
```